src/frontends/sql/decoder_context_test.cc

```diff
@@ -127,4 +135,95 @@ TEST(DecoderContextDeathTest, DecoderContextValueRegistryDeathTest) {
                "Could not find a value with id 3.");
 }
 
+class DecodeMergeOpTest
```

Comment:
What is the difference between the test here and the test with the same name  in `decode_context_test`?

Reply:
One tests the creation of the object from a proto, the other the creation of the object from input vectors. Using input vectors as inputs is more flexible and easier to test. But what I might do is create the `MergeOperator` helper class in the style that you recommended, put a `operator==` on it, and use that to remove some of the duplicated logic.

Reply:
A lot of the common logic between these tests has been moved to `testing::MergeOperationView`, so these tests should be more distinct now.